gcloud-java-contrib/gcloud-java-nio/src/main/java/com/google/gcloud/storage/contrib/nio/CloudStoragePath.java

```diff
@@ -273,9 +277,9 @@ public int compareTo(Path other) {
   public boolean equals(@Nullable Object other) {
     return this == other
         || other instanceof CloudStoragePath
-        && Objects.equals(bucket(), ((CloudStoragePath) other).bucket())
-        && Objects.equals(toRealPathInternal(false),
-            ((CloudStoragePath) other).toRealPathInternal(false));
+            && Objects.equals(bucket(), ((CloudStoragePath) other).bucket())
```

Comment:
Sorry I forgot about that one. Thanks for bringing it up. I've fixed this in #715. I also remembered what you said about tests throwing exceptions, and fixed that too :)